src/profilers.cpp

```diff
@@ -2,19 +2,375 @@
+
+    if (sanitized_metric_name.empty()) {
+        sanitized_metric_name = "metric_command_stderr.log";
+    }else{
+      sanitized_metric_name += "_stderr.log";
+    }
+
+    fs::path stderr_metric_command;
+    stderr_metric_command = result_out / sanitized_metric_name;
+
+    const int ERROR_STDERR = 201;
+    const int ERROR_STDOUT_DUP2 = 202;
+    const int ERROR_STDERR_DUP2 = 203;
+    const int ERROR_NO_NUMBER_REGEX = 204;
+    const int ERROR_TOO_MANY_NUMBERS_REGEX = 205;
+    const int ERROR_CONVERSION_TO_FLOAT = 206;
+    const int ERROR_PARSING_CONNECTION_INSTRS = 207;
+    const int ERROR_USER_REGEX_MATCH = 208;
+    const int ERROR_PIPE_METRIC_EXEC = 209;
+    
+    
+    std::future<int> metric_profiler = std::async(std::launch::async, [=]() -> int
+    {
+
+      std::vector<std::string> parts = boost::program_options::split_unix(this->metric_command);
+
+      const char *path = parts[0].c_str();
+      char *parts_c_str[parts.size() + 1];
+
+      for (int i = 0; i < parts.size(); i++) {
+        parts_c_str[i] = (char *)parts[i].c_str();
+      }
+      parts_c_str[parts.size()] = NULL;
+
+      int code_metric_exec = 0;
+
+      int status;
+      pid_t command_status = waitpid(pid, &status, WNOHANG); //check if profiled command has finished executing
+      const int interval_ns = 1000 * 1000000 / this->freq;
+
+
+      //connection write s
+      std::unique_ptr<Connection> connection;
+
+      std::regex pipe_regex(R"(pipe\s+(\d+)_(\d+))");
+      std::regex tcp_regex(R"(tcp\s+([\d\.]+)_(\d+))");
+      std::smatch instruction_match;
+      std::string connect_msg = "connect";
+
+      if (std::regex_match(instrs, instruction_match, pipe_regex)) {
+          
+          int read_fd[2];
+          int write_fd[2];
+
+          write_fd[1] = std::stoi(instruction_match[2].str());
+          read_fd[0] = std::stoi(instruction_match[1].str());
+          connection = std::make_unique<FileDescriptor>(read_fd, write_fd, this->server_buffer);
+          connection->write(connect_msg);
+
+      } else if (std::regex_match(instrs, instruction_match, tcp_regex)) {
+          
+          std::string server_address = instruction_match[1].str() + ":" + instruction_match[2].str();
+          Poco::Net::SocketAddress address(server_address);
+          Poco::Net::StreamSocket socket(address);
+
+          connection = std::make_unique<TCPSocket>(socket, this->server_buffer);
+          
+      }else{
+          // instruction is not correct
+          std::exit(ERROR_PARSING_CONNECTION_INSTRS);
+      }
+
+
+      while (command_status > 0){
+
+        struct timespec start, end;
+        clock_gettime(CLOCK_MONOTONIC, &start);
+
+        int pipe_fd[2];
+
+        if (pipe(pipe_fd) == -1) {
+          return ERROR_PIPE_METRIC_EXEC;
+        }
+
+        pid_t forked_metric_exec = fork();
+        if (forked_metric_exec == 0){
+          close(pipe_fd[0]);
+
+          int stderr_fd = creat(stderr_metric_command.c_str(), O_WRONLY);
```

Comment:
Could you replace ```O_WRONLY``` with ```S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH```?

(I know ```O_WRONLY``` is used elsewhere in the AdaptivePerf code, it's a bug which is being fixed)

Reply:
Done :+1: